src/ORM/Query.php

```diff
@@ -367,6 +375,10 @@ public function aliasField($field, $alias = null)
             $aliasedField = $alias . '.' . $field;
         }
 
+        if ($this->_autoAlias === false) {
+            return [$aliasedField];
```

Comment:
Try to build a numeric array to react on https://github.com/cakephp/cakephp/blob/master/src/Database/QueryCompiler.php#L156

Unfortunately there is another [potential problem](https://github.com/cakephp/cakephp/blob/master/src/ORM/ResultSet.php#L519) when numeric array needs to be handled. The `array_intersect_key` isn't able to work properly with `->first()`. Bring an extra condition to `ResultSet` feels a bit messy. Someone has another idea?